src/AnnotationThread.js

```diff
@@ -45,25 +49,40 @@ class AnnotationThread extends EventEmitter {
         super();
 
         this.annotatedElement = data.annotatedElement;
-        this.annotations = data.annotations || {};
-        this.annotationService = data.annotationService;
+        this.api = data.api;
         this.container = data.container;
         this.fileVersionId = data.fileVersionId;
         this.location = data.location;
-        this.threadID = data.threadID || AnnotationService.generateID();
+        this.threadID = data.threadID || AnnotationAPI.generateID();
         this.threadNumber = data.threadNumber || '';
         this.type = data.type;
         this.locale = data.locale;
-        this.isMobile = data.isMobile || false;
         this.hasTouch = data.hasTouch || false;
         this.permissions = data.permissions;
-        this.localized = data.localized;
         this.state = STATES.inactive;
+        this.canComment = true;
+        this.headerHeight = data.headerHeight;
 
-        this.regenerateBoundary();
+        this.id = data.id;
```

Comment:
There's a ton of common data between here and Annotation.js ( fileVersionId, createdBy, modifiedAt, createdAt, etc.). Would it be possible to make a utility method that extracts the common data for easier assignment?

Something along the lines of:
```
function appendCommonAnnotationData(object, data) {
  // extract out common props we want
  const commonData = Object.assign({}, COMMON_ANNOTATION_SCHEMA, data);
  // and then apply it to the object that requires the data however you want to
  Object.keys(commonData).forEach((key) => object[key] = commonData[key])
}
```

Reply:
Actually meant to delete Annotation.js! Doing that now :)